api/storage_service.cc

```diff
@@ -127,14 +128,14 @@ seastar::future<json::json_return_type> run_toppartitions_query(db::toppartition
 
                 for (auto& d: topk_results.read.top(q.list_size())) {
                     cf::toppartitions_record r;
-                    r.partition = (legacy_request ? "" : "(" + d.item.schema->ks_name() + ":" + d.item.schema->cf_name() + ") ") + sstring(d.item);
+                    r.partition = (legacy_request ? "" : "(" + d.item.schema->ks_name() + ":" + d.item.schema->cf_name() + ") ") + utils::utf8::escape_json(sstring(d.item));
                     r.count = d.count;
                     r.error = d.error;
                     results.read.push(r);
                 }
                 for (auto& d: topk_results.write.top(q.list_size())) {
                     cf::toppartitions_record r;
-                    r.partition = (legacy_request ? "" : "(" + d.item.schema->ks_name() + ":" + d.item.schema->cf_name() + ") ") + sstring(d.item);
+                    r.partition = (legacy_request ? "" : "(" + d.item.schema->ks_name() + ":" + d.item.schema->cf_name() + ") ") + utils::utf8::escape_json(sstring(d.item));
                     r.count = d.count;
                     r.error = d.error;
                     results.write.push(r);
```

Comment:
I think that json serialization should automatically do this, rather than having to remember to escape every time we add data to a json_return_type. Please check where this serialization happens (probably in Seastar) and fix it there.

Reply:
That's what I wanted to avoid...
Lots of client apps might rely on the jsons as they are now. Toppartitions is fairly new, so I won't break much anyway.
If you're sure about this I'll do it, but please confirm.